skglm/penalties/non_separable.py

```diff
@@ -40,14 +40,16 @@ def value(self, w):
         return np.sum(np.sort(np.abs(w)) * self.alphas[::-1])
 
     def prox_vec(self, x, stepsize):
-        def _prox(_x, _alphas):
-            sign_x = np.sign(_x)
-            _x = np.abs(_x)
-            sorted_indices = np.argsort(_x)[::-1]
-            prox = prox_SLOPE(_x[sorted_indices], _alphas)
-            prox[sorted_indices] = prox
```

Comment:
@mathurinm, found the damn bug :bug:! 

it is precisely this line. In particular, refer to example in debug script, ``numpy`` interpret it as

```python
prox[2] = prox[0]
prox[0] = prox[1]
prox[1] = prox[2]
```
and hence overrides `prox[2]`. We endup thereafter with `prox[1] == prox[0]`

Reply:
I believe when compiling, we move to the realm of numba, where the intermediate variable are created automatically hence not having the error.